src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/attn_quant.cpp

```diff
@@ -307,12 +485,116 @@ static void paged_attn_quant_mt(const ov::intel_cpu::PlainTensor& k_src,
                                 const ov::intel_cpu::PlainTensor& v_src,
                                 const ov::intel_cpu::PlainTensor& k_dst,
                                 const ov::intel_cpu::PlainTensor& v_dst,
+                                const ov::intel_cpu::PlainTensor& past_lens,
+                                const ov::intel_cpu::PlainTensor& subsequence_begins,
+                                const ov::intel_cpu::PlainTensor& block_indices,
+                                const ov::intel_cpu::PlainTensor& block_indices_begins,
                                 const ov::intel_cpu::PlainTensor& slot_mapping,
+                                ov::intel_cpu::PlainTensor& temp_buffer,
+                                const bool quant_key_by_channel,
                                 const size_t key_group_size,
                                 const size_t value_group_size) {
     size_t B = k_src.m_dims[0], H = k_src.m_dims[1], L1 = k_src.m_dims[2], S = k_src.m_dims[3], SV = v_src.m_dims[3];
-    size_t block_size = k_dst.m_dims[2];
+    size_t block_size = quant_key_by_channel ? k_dst.m_dims[2] - 2 * sizeof(float) : k_dst.m_dims[2];
     size_t sub_byte_multiplier = 8 / v_dst.get_precision().bitwidth();
+    if (quant_key_by_channel) {
+        parallel_for2d(past_lens.size(0), H, [&](size_t sub_seq_id, size_t h) {
+            auto past_len = past_lens.ptr<int32_t>()[sub_seq_id];
+            float* buffer = temp_buffer.ptr<float>(parallel_get_thread_num());
+            auto q_len =
+                subsequence_begins.ptr<int32_t>()[sub_seq_id + 1] - subsequence_begins.ptr<int32_t>()[sub_seq_id];
+            auto block_number_start = block_indices_begins.ptr<int32_t>()[sub_seq_id];
+            auto total_blocks =
+                block_indices_begins.ptr<int32_t>()[sub_seq_id + 1] - block_indices_begins.ptr<int32_t>()[sub_seq_id];
+            size_t m = 0;
+            if (past_len == 0) {
+                parallel_for(total_blocks, [&](int32_t block_count) {
+                    auto block_id = block_number_start + block_count;
+                    auto block_number = block_indices.ptr<int32_t>()[block_id];
+                    auto token_num = (block_id == (block_indices_begins.ptr<int32_t>()[sub_seq_id + 1] - 1))
+                                         ? (q_len - (block_id - block_number_start) * block_size)
```

Comment:
should be `(q_len - block_count * block_size)`?

Reply:
block_count == block_id - block_number_start, but yes `block_count` is more readable

Reply:
fixed